### PR TITLE
[Serve] Add metric to track ongoing requests in proxies

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change adds a metric (`"serve_num_ongoing_requests"`) to track the number of ongoing requests in each HTTP Proxy.

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/http_util.py [hunk 1]
```diff
@@ -456,3 +457,37 @@ def validate_http_proxy_callback_return(
...
+    This class keeps items in an internal set, and it updates a Prometheus
+    gauge with the the set's size whenever the `add`, `remove`, or `discard`
+    methods are called.
+    """
+
+    def __init__(self, connection_gauge: metrics.Gauge):
+        self.connection_gauge = connection_gauge
+        self.internal_set = set()
+
+    def add(self, element: Any):
```
Comment (GeneDer) [inline python/ray/serve/_private/http_util.py:478]: Non-blocker: the element should probably be a protocol type https://github.com/encode/uvicorn/blob/master/uvicorn/server.py#L44

**Review by GeneDer — COMMENTED** [review]
I think this is a very cool idea, and does the job. Alternatively can probably just follow closely with [_ongoing_requests](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/http_proxy.py#L302), but I think your implementation will be slightly more up-to-date.

**Comment by shrekris-anyscale** [comment]
@GeneDer I switched this PR to use `ongoing_requests` as you recommended. Please take a look!

**Review by GeneDer — APPROVED** [review]
LGTM

**Comment by shrekris-anyscale** [comment]
Sorry, bad rebase. I'll close this and open a new PR.

**Pull request closed without merging**